ml_server/main.py

```diff
@@ -0,0 +1,79 @@
+import os
+from pathlib import Path
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
+from typing import List
+from functools import lru_cache
+
+from fastapi import Depends
+from gliner import GLiNER
+
+from tenant_dependency import get_tenant_info
+
+KEYCLOAK_HOST = os.getenv("KEYCLOAK_HOST", "")
+TENANT = get_tenant_info(url=KEYCLOAK_HOST, algorithm="RS256")
+ROOT_PATH = os.environ.get("ROOT_PATH", default="")
+
+
+# Cache will save up to this number of most recently used models. Larger amount will consume more memory,
+# but allows more models to be initialized at the same time.
+NUM_MODELS_TO_CACHE = 3
```

Comment:
Not sure if we need this comment, var name self explanatory.
Maybe name it MODELS_CACHE_SIZE?

Reply:
This comment is more for explaining why this var exists and what will happen if someone want to change it, just in case.